java/arcs/core/data/proto/manifest.proto

```diff
@@ -143,11 +143,15 @@ message ReferenceTypeProto {
   TypeProto referred_type = 1;
 }
 
-// TODO(b/155944755) more fields pending.
 message TypeVariableProto {
-  // Identifies a type variable in a recipe or a particle spec.
-  // Type variables with the same name are linked together.
   string name = 1;
+  ConstraintInfo constraint = 2;
+}
+
+// These are constraints expressed by the particle author,
+// not those resulting from type inference.
+message ConstraintInfo {
+  TypeProto constraint_type = 1;
```

Comment:
@piotrswigon Weren't we also about `~a with {*}`? How can we represent that?

Reply:
Let's wait for BAR today to discuss which direction we pursue.

My gut feeling is that we should not try to represent `{*}` as an Arcs type - as it does not match anything the type system can express (maybe a TV that's bound by another TV that's not bound at all?), instead I would argue for `~a with {*}` to be translated into `TypeVariable {isReflective = true}`, maybe s/reflective/introspective/` to something like that.